src/sources/novelfull.py

```diff
@@ -91,19 +93,20 @@ def download_chapter_list(self, page):
 
         for a in soup.select('ul.list-chapter li a'):
             title = a['title'].strip()
-
+            prev_ch_vol = self.chapters[-1]['volume'] if self.chapters else None
             chapter_id = len(self.chapters) + 1
-            match = re.findall(r'ch(apter)? (\d+)', title, re.IGNORECASE)
-            if len(match) == 1:
-                chapter_id = int(match[0][1])
-            # end if
+            volume_id = (chapter_id - 1) // 100 + 1
+            ch_match = re.search(RE_CHAPTER, title, re.IGNORECASE)
+            vol_match = re.search(RE_VOLUME, title, re.IGNORECASE)
 
-            volume_id = 1 + (chapter_id - 1) // 100
-            match = re.findall(r'(book|vol|volume) (\d+)',
-                               title, re.IGNORECASE)
-            if len(match) == 1:
-                volume_id = int(match[0][1])
-            # end if
+            if ch_match:
+                chapter_id = int(ch_match.group(2))
+
+            _v_id = vol_match.group(1) if vol_match else None
+            if _v_id and (
+                not prev_ch_vol or _v_id == prev_ch_vol or _v_id == prev_ch_vol + 1
```

Comment:
this uses the matched index if `self.chapters` is empty or if the matched vol is the same or +1 as previous `chapters.volume`

Reply:
nicely done